### PR TITLE
Migrate react-ui & tools from jest to vitest

### PR DESCRIPTION
**Pull request by webpro**
This PR migrates react-ui and the tools app to Vitest (the docs app is cleaned up, does not have such tests).

This was an interesting migration, since the combo of Vitest + happy-dom results in actually parsing the stylesheets, ~which led me to find bugs (regarding the dark theme class) that also have been fixed in this PR.~

The ESLint plugins we can reconsider when migrations are finished. For now it's removed as it wasn't doing anything.

Some mocks have been removed, an a test is un-skipped, resulting in increased coverage.

### CODE REVIEW

File: packages/apps/docs/src/components/PageHelpful/styles.css.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const textAreaClass = style([
     '::placeholder': {
       color: vars.colors.$gray40,
     },
-    [`${darkThemeClass} &::placeholder`]: {
+    [`.${darkThemeClass} &::placeholder`]: {
```
Comment (sstraatemans) [inline packages/apps/docs/src/components/PageHelpful/styles.css.ts:26]: this is not needed.

**Review by sstraatemans — CHANGES_REQUESTED** [review]
the darkthemeclass does not need the '.'

File: packages/apps/docs/src/components/PageHelpful/styles.css.ts (hunk 1 above)
Comment (webpro) [reply packages/apps/docs/src/components/PageHelpful/styles.css.ts:26]: The class name variable does not and should not contain a dot (a selector might contain one) and the stylesheet parser in our tests complains about it. Maybe there's something that fixes it in the chain eventually, but when parsing the stylesheet happy-dom rightfully complained about it.

Comment (webpro) [reply packages/apps/docs/src/components/PageHelpful/styles.css.ts:26]: Well, this is embarrassing. I've been analyzing stylesheets spitted out by the stylesheet parser from happy-dom, and found those dots missing. Now I can no longer reproduce this, so the dots are back. Apologies!

File: packages/libs/react-ui/src/styles/sprinkles.css.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ const systemProperties = defineProperties({
 const colorProperties = defineProperties({
   conditions: {
     lightMode: {},
-    darkMode: { selector: `.${darkThemeClass} &` },
+    darkMode: { selector: `${darkThemeClass} &` },
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/styles/sprinkles.css.ts:36]: We should leave the `.` in front of the classname. The variable only includes the class name

**Review by sstraatemans — APPROVED** [review]

File: packages/apps/docs/src/components/PageHelpful/styles.css.ts (hunk 1 above)
Comment (eileenmguo) [reply packages/apps/docs/src/components/PageHelpful/styles.css.ts]: Yeah, the variable shouldn't include the dot, so adding the dot is necessary

File: packages/libs/react-ui/src/styles/sprinkles.css.ts (hunk 2 above)
Comment (webpro) [reply packages/libs/react-ui/src/styles/sprinkles.css.ts]: OK restored this too. It's the same variable, no? 🤷  Oh, well.

Comment (eileenmguo) [reply packages/libs/react-ui/src/styles/sprinkles.css.ts]: Yes, it is the same variable haha

**Review by timoheddes — APPROVED** [review]

**Review by alber70g — APPROVED** [review]

File: packages/libs/react-ui/src/styles/sprinkles.css.ts (hunk 2 above)
Comment (webpro) [reply packages/libs/react-ui/src/styles/sprinkles.css.ts]: So this PR no longer has changes related to this, and the migrated tests are passing, so let's leave it at this for now.

File: packages/apps/docs/src/components/PageHelpful/styles.css.ts (hunk 1 above)
Comment (webpro) [reply packages/apps/docs/src/components/PageHelpful/styles.css.ts]: So this PR no longer has changes related to this, and the migrated tests are passing, so let's leave it at this for now.

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**